src/txn/txn_recover.c

```diff
@@ -688,6 +690,18 @@ __wt_txn_recover(WT_SESSION_IMPL *session)
 done:
     WT_ERR(__recovery_set_checkpoint_timestamp(&r));
 
+    /*
+     * Check whether the history store file exists or not. If it does not, then we should not apply
+     * rollback to stable to each table and we should not perform a checkpoint. This might happen if
+     * we're upgrading from an older version.
+     */
+    ret = __wt_hs_cursor(session, &session_flags, &is_owner);
```

Comment:
This seems a bit nasty so I was considering using some lower-level API to check whether the file exists. However, we're going to use the history store cursor as part of rollback to stable so it kind of makes sense to open it and leave it open for the remainder of the function.

Reply:
I think we need to consider the case when the history store file is corrupted as well.

Reply:
We thought of that. But what indicator do we have that we're upgrading vs us having a corrupted history store?

Reply:
> I think we need to consider the case when the history store file is corrupted as well.

I think we should treat that separately. This is for the common case of upgrading from an earlier version of MongoDB when the file isn't there. Can we open a ticket to handle a corrupted history store (when the corruption is cheaply detectable.